libcxx/include/complex

```diff
@@ -892,7 +790,7 @@ typename enable_if
 >::type
 arg(_Tp __re)
 {
-    return atan2(0., __re);
+    return atan2(0., static_cast<double>(__re));
```

Comment:
Why is this needed?

Reply:
This is an error from when long double was shunted through `complex<double>`  in order to satisfy math intrinsics on device.

Reply:
Well, apparently that wasn't the cause. There's no integral `atan2` in device code and I think the signature prevents promotion to double somehow. Do you have advice on a solution? I can split the integral path from double and make it more explicit that way.